File: code/Diana0.01/model/Model/workflownodeaction.cs
using System;
using System.Linq;
using System.Text;
/// <summary>
///项目名称：Diana轻量级开发框架
///版本：0.0.1版
///开发团队成员：胡凯雨，张梦丽，艾美珍，易传佳，陈祚送，康文洋，张婷婷，王露，周庆，夏萍萍，陈兰兰
///模块和代码页功能描述：工作流节点动作实体类
///最后修改时间：2018/1/26
/// </summary>
namespace Diana.model
{
    public class workflownodeaction
    {
        /// <summary>
        /// Desc:- 
        /// Default:- 
        /// Nullable:False 
        /// </summary>
        public string nodeactionname { get; set; }
        /// <summary>
        /// Desc:- 
        /// Default:- 
        /// Nullable:False 
        /// </summary>
        public string nodeactionmemo { get; set; }
        /// <summary>
        /// Desc:- 
        /// Default:- 
        /// Nullable:False 
        /// </summary>
        public int? nodetype { get; set; }
        /// <summary>
        /// Desc:- 
        /// Default:- 
        /// Nullable:False 
        /// </summary>
        public int currentnodeid { get; set; }

        /// <summary>
        /// Desc:- 
        /// Default:- 
        /// Nullable:False 
        /// </summary>
        public int nextnodeid { get; set; }

        /// <summary>
        /// Desc:- 
        /// Default:- 
        /// Nullable:False 
        /// </summary>
        public int? nastatus { get; set; }

        /// <summary>
        /// Desc:- 
        /// Default:- 
        /// Nullable:False 
        /// </summary>
        public int id { get; set; }

        /// <summary>
        /// Desc:- 
        /// Default:- 
        /// Nullable:False 
        /// </summary>
        public int? wfid { get; set; }

        /// <summary>
        /// Desc:- 
        /// Default:- 
        /// Nullable:False 
        /// </summary>
        public DateTime? begintime { get; set; }

        /// <summary>
        /// Desc:- 
        /// Default:- 
        /// Nullable:False 
        /// </summary>
        public DateTime? endtime { get; set; }

        /// <summary>
        /// Desc:- 
        /// Default:- 
        /// Nullable:False 
        /// </summary>
        public string nacondition { get; set; }

        /// <summary>
        /// Desc:- 
        /// Default:- 
        /// Nullable:False 
        /// </summary>
        public int? nalock { get; set; }
        /// <summary>
        /// Desc:- 
        /// Default:- 
        /// Nullable:False 
        /// </summary>
        public int? nodeactioncode { get; set; }
        /// <summary>
        /// Desc:- 
        /// Default:- 
        /// Nullable:False 
        /// </summary>
        public int? actionid { get; set; }
        
    }
}

File: code/Diana0.01/Web/Templates/BootstrapTemplate/js/workflownodeaction.js
$(function () {
        gridList();
    })
    function gridList() {
        var $gridList = $("#gridList");
        $gridList.dataGrid({
            url: "../index.ashx?ctrl=workflownodeaction&action=showAll",
            height: $(window).height() - 126,
            colModel: [
                { label: "主键", name: "id", hidden: true, key: true },
                {
                    label: '工作流', name: 'wfid', width: 150, align: 'left',
                    formatter: function (cellvalue) {
                        $.ajax({
                            url: "../index.ashx?ctrl=workflownodeaction&action=GetSelectJson",
                            data: { keyValue: cellvalue },
                            dataType: "json",
                            async: false,
                            success: function (data) {
                                cellvalue = data[0].text;
                            }
                        });
                        return cellvalue;
                    }
                },
                { label: '节点动作名称', name: 'nodeactionname', width: 150, align: 'left' },
                {
                    label: '当前节点', name: 'currentnodeid', width: 150, align: 'left',
                    formatter: function (cellvalue) {
                        $.ajax({
                            url: "../index.ashx?ctrl=workflownodeaction&action=GetSelectNodeJson",
                            data: { keyValue: cellvalue },
                            dataType: "json",
                            async: false,
                            success: function (data) {
                                cellvalue = data[0].text;
                            }
                        });
                        return cellvalue;
                    }
                },
                {
                    label: '下一个节点', name: 'nextnodeid', width: 150, align: 'left',
                    formatter: function (cellvalue) {
                        $.ajax({
                            url: "../index.ashx?ctrl=workflownodeaction&action=GetSelectNodeJson",
                            data: { keyValue: cellvalue },
                            dataType: "json",
                            async: false,
                            success: function (data) {
                                cellvalue = data[0].text;
                            }
                        });
                        return cellvalue;
                    }
                },
                {
                    label: '动作状态', name: 'nastatus', width: 150, align: 'left',
                    formatter: function (cellvalue) {
                        return cellvalue == 1 ? "启用" : "未启用";
                    }
                },
                { label: '开始时间', name: 'begintime', width: 150, align: 'left' },
                { label: ' 结束时间 ', name: 'endtime', width: 150, align: 'left' },
                {
                    label: '锁定状态', name: 'nalock', width: 150, align: 'left',
                    formatter: function (cellvalue) {
                        return cellvalue == 1 ? "锁定" : "未锁定";
                    }
                },
                {
                    label: '动作类型', name: 'nodetype', width: 150, align: 'left',
                    formatter: function (cellvalue) {
                        switch (cellvalue) {
                            case 1:
                                cellvalue = "未定义1";
                                break;
                            case 2:
                                cellvalue = "未定义2";
                                break;
                            default:
                                cellvalue = "未定义";
                                break;
                        }
                        return cellvalue;
                    }
                }

            ],
            pager: "#gridPager",
            sortname: 'id asc',
            viewrecords: true
        });
        $("#btn_search").click(function () {
            $gridList.jqGrid('setGridParam', {
                postData: { keyword: $("#txt_keyword").val() },
            }).trigger('reloadGrid');
        });
    }
    function btn_add() {
        $.modalOpen({
            id: "Form",
            title: "新增节点动作",
            url: "../index.ashx?ctrl=workflownodeaction&action=Form",
            width: "700px",
            height: "510px",
            callBack: function (iframeId) {
                top.frames[iframeId].submitForm();
            }
        });
    }
    function btn_edit() {
        var keyValue = $("#gridList").jqGridRowValue().id;
        $.modalOpen({
            id: "Form",
            title: "修改节点动作",
            url: "../index.ashx?ctrl=workflownodeaction&action=Form&keyValue=" + keyValue,
            width: "700px",
            height: "510px",
            callBack: function (iframeId) {
                top.frames[iframeId].submitForm();
            }
        });
    }
    function btn_delete() {
        $.deleteForm({
            url: "../index.ashx?ctrl=workflownodeaction&action=DeleteForm",
            param: { keyValue: $("#gridList").jqGridRowValue().id },
            success: function () {
                $.currentWindow().$("#gridList").trigger("reloadGrid");
            }
        })
    }
    function btn_details() {
        var keyValue = $("#gridList").jqGridRowValue().id;
        $.modalOpen({
            id: "Details",
            title: "查看节点动作",
            url: "../index.ashx?ctrl=workflownodeaction&action=Details&keyValue=" + keyValue,
            width: "700px",
            height: "510px",
            btn: null,
        });
    }
    function btn_copyandpaste() {
        $.deleteForm({
            url: "../index.ashx?ctrl=workflownodeaction&action=copyAndPasteForm",
            param: { keyValue: $("#gridList").jqGridRowValue().id },
            success: function () {
                $.currentWindow().$("#gridList").trigger("reloadGrid");
            }
        })
    }